lib/ts/recipe/thirdparty/types.ts

```diff
@@ -46,35 +46,33 @@ export type NormalisedInputType = AuthRecipeNormalisedInputType<PreAndPostAPIHoo
 };
 
 export type StateObject = {
-    state?: string;
-    rid?: string;
+    stateForAuthProvider?: string;
     thirdPartyId?: string;
```

Comment:
This should not be optional. And change `thirdPartyId` to `providerId` everywhere